qos-service-impl/src/main/java/com/palantir/atlasdb/qos/ratelimit/QosRateLimiter.java

```diff
@@ -79,13 +80,13 @@ public Duration consumeWithBackoff(int estimatedNumUnits) {
     }
 
     /**
-     * Records an adjustment to the original estimate of units consumed passed to {@link #consumeWithBackoff(int)}. This
+     * Records an adjustment to the original estimate of units consumed passed to {@link #consumeWithBackoff}. This
      * should be called after a query returns, when the exact number of units consumed is known. This value may be
      * positive (if the original estimate was too small) or negative (if the original estimate was too large).
      */
-    public void recordAdjustment(int adjustmentUnits) {
+    public void recordAdjustment(long adjustmentUnits) {
         if (adjustmentUnits > 0) {
-            rateLimiter.steal(adjustmentUnits);
+            rateLimiter.steal(Ints.saturatedCast(adjustmentUnits)); // TODO(nziebart): deal with longs
         }
         // TODO(nziebart): handle negative case
```

Comment:
are these TODOs tracked somewhere outside this code?
(I guess you can just grep the qos-* directories for your username, so not a bit deal...)